far/stddlg.cpp

```diff
@@ -76,6 +76,69 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 //----------------------------------------------------------------------------
 
+
+namespace
+{
+	template<SearchReplaceDlgParams::SharedGroup group>
```

Comment:
Looks like templatizing doesn't contribute anything here except producing unique instances.
Maybe we can avoid templates completely with something like

```C++
auto& GetShared(SharedGroup const Id)
{
    static SearchReplaceDlgParams s_SharedParams[SharedGroup::count];
    return s_SharedParams[Id];
}
```
?

I normally wouldn't care since templatized parts are already minimal, but those explicit instantiations below make things look more complex than they could be.

Reply:
Well, using templates here has some advantages:
- Logical: instantiation of the shared params of different groups are not tied together. In this case it's irrelevant, but in general we may want to avoid instantiating `help` params only because somebody needs `common` params.
- Performance: no indexing, no indirections.
- Encapsulation: no implementation details, like `count`, is exposed to the clients.

Each of these considerations are, of course, minor. However, all of them together made my choice, especially since I do not see any extra complexity in explicit template specialization.

Anyway, I change these functions to use indices.